amf-webapi/shared/src/main/scala/amf/plugins/domain/shapes/resolution/stages/merge/JsonMergePatchAlgorithm.scala

```diff
@@ -6,9 +6,13 @@ import amf.core.model.domain.{AmfArray, AmfElement, AmfObject, AmfScalar}
 import amf.core.parser.FieldEntry
 import amf.plugins.domain.webapi.models.{Key, Payload}
 
-case class JsonMergePatch(isNull: AmfElement => Boolean, keyCriteria: KeyCriteria, ignoredFields: Seq[Field] = Seq()) {
+case class JsonMergePatch(isNull: AmfElement => Boolean,
+                          keyCriteria: KeyCriteria,
+                          ignoredFields: Seq[Field] = Seq(),
+                          customMerges: Seq[CustomMerge] = Seq()) {
 
   def merge[T <: AmfElement](target: T, patch: T): AmfElement = {
+    customMerges.foreach(_.apply(target, patch))
```

Comment:
Would probably skip them from the merge altogether, filtering the Field by both the ignoredFields and if it applies for a CustomMerge